python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/arp.j2

```diff
@@ -17,14 +17,7 @@
 arp aging timeout default {{ arp.aging.timeout_default }}
 {%     endif %}
 {%     if arp.static_entries is arista.avd.defined %}
-{# TODO - remove when dropping support for ansible 2.12.6 and its jinja2 hack #}
-{%         for entry in arp.static_entries %}
-{%             if entry.vrf is not arista.avd.defined %}
-{%                 do entry.update({"vrf": "default"}) %}
-{%             endif %}
-{%         endfor %}
-{%         for vrf, entries in arp.static_entries | groupby("vrf") | arista.avd.natural_sort %}
-{# TODO {%     for vrf, entries in arp.static_entries | arista.avd.default([]) | groupby("vrf", default="default") | arista.avd.natural_sort %} #}
+{%         for vrf, entries in arp.static_entries | arista.avd.default([]) | groupby("vrf", default="default") | arista.avd.natural_sort %}
```

Comment:
no need for the default([]) since we already check for static_entried in the line above.